### PR TITLE
Update build system to use buildx, support multi-arch builds

### PR DESCRIPTION
**Pull request by Chuxel**
Implements https://github.com/microsoft/vscode-dev-containers/issues/914. 

This PR does not update config to start building linux/arm64, but rather sets up the build system to do so. See https://github.com/microsoft/vscode-dev-containers/issues/558#issuecomment-905117722 for issues preventing switch, current plan of action.

### CODE REVIEW

File: build/src/push.js [hunk 1]
```diff
@@ -30,6 +30,16 @@ async function push(repo, release, updateLatest, registry, registryPath, stubReg
...
 
+    // Use or create builder
+    console.log('(*) Setting up builder...');
+    const builders = await asyncUtils.exec('docker buildx ls');
+    if(builders.indexOf('vscode-dev-containers') < 0) {
+        await asyncUtils.spawn('docker', ['buildx', 'create', '--use', '--name', 'vscode-dev-containers']);
+    } else {
+        await asyncUtils.spawn('docker', ['buildx', 'use', 'vscode-dev-containers']);
+    }
+    await asyncUtils.spawn('docker', ['run', '--privileged', '--rm', 'tonistiigi/binfmt', '--install', 'all']);
```
Comment (joshspicer) [inline build/src/push.js:41]: No idea what this line is doing :D .  Maybe add a comment to make it more clear?

File: build/src/push.js [hunk 2]
```diff
@@ -84,7 +94,26 @@ async function pushImage(definitionId, repo, release, updateLatest,
...
+                case 'undefined': architectures = ['linux/amd64']; break;
+            }
+            console.log(`(*) Target image architectures: ${architectures.reduce((prev, current) => prev += `\n     ${current}`, '')}`);
+            let localArchitecture = process.arch;
+            switch(localArchitecture) {
+                case 'arm': localArchitecture = 'linux/arm/v7'; break;
+                case 'aarch32': localArchitecture = 'linux/arm/v7'; break;
+                case 'aarch64': localArchitecture = 'linux/arm64'; break;
+                case 'x64': localArchitecture = 'linux/amd64'; break;
+                case 'x32': localArchitecture = 'linux/386'; break;
```
Comment (joshspicer) [inline build/src/push.js:110]: is this supposed to be `linux/i386`?  Doesn't look right to me without the `i` for some reason, but I might just be wrong

**Review by joshspicer — APPROVED** [review]

File: build/src/push.js (hunk 1 above)
Comment (Chuxel) [reply build/src/push.js:41]: Sure - it sets up QEMU to enable multi-platform builds. Docker Desktop does htis as a part of install so people don't know. Will add a comment.

File: build/src/push.js (hunk 2 above)
Comment (Chuxel) [reply build/src/push.js:110]: Yeah for Docker it's w/o the `i`.  Type `docker buildx ls` locally to see architectures that are setup to emulate. Desktop has them all installed.

**Comment by Chuxel** [comment]
@joshspicer Triggered dev builds and they all successfully ran. The codespaces-linux image will now be built using buildx which should help reduce total layers.

**Pull request merged**